src/defaultCharDataLoader.js

```diff
@@ -4,17 +4,24 @@
 const VERSION = '1';
 const getCharDataUrl = (char) => `https://cdn.jsdelivr.net/npm/hanzi-writer-data@${VERSION}/${char}.json`;
 
-module.exports = (char, onLoad) => {
+module.exports = (char, onLoad, onError) => {
   // load char data from hanziwriter.org cdn (currently hosted on github pages)
   const xhr = new global.XMLHttpRequest();
   if (xhr.overrideMimeType) { // IE 9 and 10 don't seem to support this...
     xhr.overrideMimeType('application/json');
   }
   xhr.open('GET', getCharDataUrl(char), true);
+  xhr.onerror = (event) => {
+    onError(xhr, event);
+  };
   xhr.onreadystatechange = () => {
     // TODO: error handling
-    if (xhr.readyState === 4 && xhr.status === 200) {
+    if (xhr.readyState !== 4) return;
+
+    if (xhr.status === 200) {
       onLoad(JSON.parse(xhr.responseText));
+    } else if (onError) {
+      onError(xhr);
```

Comment:
👍

Reply:
did you notice that the ``onError`` callback gets called in 3 different occasions.
- once by might be by ``_loadCharacter`` implementation whenever a new loading is asked and the previous is not finished.
- and twice in the ``defaultCharDataLoader``: in case of ``XmlHttpRequest.onerror`` event, and if the request is successfully answered but the HTTP error status in not '200'.
If you can distinguish from these 3 cases, it still one unique function getting called with 3 different way (different args and different ``this``). I didn't like that much, but I figured it gave a great deal of tinkerability without adding to much callbacks to deal with for very similar problems in the outer API of your library.

Reply:
Hrm that's a good point. I think the 2 calls in `defaultCharDataLoader` are fine, since they both pass the `xhr` object back for the developer to figure out what happened. Probably for `_loadCharacter` when the previous request isn't finished, it seems like that shouldn't be an error. The new character should just take precedence and the old loading should be ignored, I think. Maybe we can pass some parameters in the reject to indicate if it's being rejected because of a real error, and just ignore the reject otherwise?

Reply:
Yup, I like this suggestion much better. I'll do that.